cni/pkg/plugin/plugin.go

```diff
@@ -227,7 +227,7 @@ func CmdAdd(args *skel.CmdArgs) (err error) {
 		log.Infof("ambientConf.Mode: %s", ambientConf.Mode)
 		log.Infof("ambientConf.ZTunnelReady: %v", ambientConf.ZTunnelReady)
 		added := false
-		if ambientConf.Mode != ambient.AmbientMeshOff.String() && ambientConf.ZTunnelReady {
+		if !excludePod && ambientConf.Mode != ambient.AmbientMeshOff.String() && ambientConf.ZTunnelReady {
```

Comment:
nit: you can remove the check on L242 then

Reply:
OK, will open another PR after this been merged.